File: katalyze/src/messageHandlers/MessageHandler.java
package messageHandlers;

import io.SimpleMessage;
import model.Contest;

public interface MessageHandler {

	public abstract void connectTo(Contest contest) throws Exception;

	public abstract Boolean supports(SimpleMessage message);
	
	public abstract void close() throws Exception;
	
	public abstract void process(SimpleMessage message) throws Exception;

}

File: katalyze/src/main/kotlin/messageHandlers/MessageHandler.kt
package messageHandlers

import legacyfeed.SimpleMessage
import model.Contest

interface MessageHandler {
    @Throws(Exception::class)
    fun connectTo(contest: Contest)

    fun supports(message: SimpleMessage): Boolean

    @Throws(Exception::class)
    fun close()

    @Throws(Exception::class)
    fun process(message: SimpleMessage)
}